server/services/appointmentService.ts

```diff
@@ -38,42 +40,54 @@ export default class AppointmentService {
-          appointmentTypeDescription: apt.customName
-            ? `${apt.customName} (${apt.category.description})`
-            : apt.category.description,
-          locationId: apt.internalLocation?.id,
-          locationDescription: apt.internalLocation?.description || (apt.inCell ? 'In cell' : undefined),
-          status: apt.isCancelled ? 'CANCELLED' : 'ACTIVE',
-          viewAppointmentLink: `${config.activitiesAndAppointmentsUrl}/appointments/${apt.appointmentId}`,
-          createdTime: apt.createdTime,
-          updatedTime: apt.updatedTime,
-          cancelledTime: apt.cancelledTime,
-          cancelledBy: apt.cancelledBy,
-        })),
-      ),
-    )
+    return this.activitiesAndAppointmentsApiClient
+      .getAppointments(prisonId, { date, timeSlots: periods }, user)
+      .then(appts =>
+        appts.flatMap(apt =>
+          apt.attendees.map(a => ({
+            id: apt.appointmentId,
+            offenderNo: a.prisonerNumber,
+            date: apt.startDate,
+            startTime: apt.startTime,
+            endTime: apt.endTime,
+            appointmentTypeCode: apt.category.code,
+            appointmentTypeDescription: apt.customName
+              ? `${apt.customName} (${apt.category.description})`
+              : apt.category.description,
+            locationId: apt.internalLocation?.id,
+            locationDescription: apt.internalLocation?.description || (apt.inCell ? 'In cell' : undefined),
+            status: apt.isCancelled ? 'CANCELLED' : 'ACTIVE',
+            viewAppointmentLink: `${config.activitiesAndAppointmentsUrl}/appointments/${apt.appointmentId}`,
+            createdTime: apt.createdTime,
+            updatedTime: apt.updatedTime,
+            cancelledTime: apt.cancelledTime,
+            cancelledBy: apt.cancelledBy,
+          })),
+        ),
+      )
   }
 
-  private async getAppointmentsFromPrisonApi(prisonId: string, date: Date, user: Express.User): Promise<Appointment[]> {
-    return this.prisonApiClient.getAppointments(prisonId, date, user).then(appts =>
-      appts.map(a => ({
-        ...a,
-        startTime: formatDate(a.startTime, 'HH:mm'),
-        endTime: formatDate(a.endTime, 'HH:mm'),
-        status: 'ACTIVE', // All appointments returned by Prison API are active appointments
-        viewAppointmentLink: `${config.dpsUrl}/appointment-details/${a.id}`,
-      })),
-    )
+  private async getAppointmentsFromPrisonApi(
+    prisonId: string,
+    date: Date,
+    periods: Period[],
+    user: Express.User,
+  ): Promise<Appointment[]> {
+    const periodList = periods ?? [undefined]
+    return Promise.all(periodList.map(p => this.prisonApiClient.getAppointments(prisonId, date, p, user)))
+      .then(appts => appts.flat())
+      .then(appts => _.uniq(appts))
+      .then(appts =>
+        appts.map(a => ({
+          ...a,
+          startTime: formatDate(a.startTime, 'HH:mm'),
+          endTime: formatDate(a.endTime, 'HH:mm'),
```

Comment:
Although they should have an end date, I think appointments from prison-api can actually have null end dates.